python/paddle/fluid/contrib/slim/tests/test_mkldnn_int8_quantization_strategy.py

```diff
@@ -0,0 +1,220 @@
+#   copyright (c) 2019 paddlepaddle authors. all rights reserved.
+#
+# licensed under the apache license, version 2.0 (the "license");
+# you may not use this file except in compliance with the license.
+# you may obtain a copy of the license at
+#
+#     http://www.apache.org/licenses/license-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the license is distributed on an "as is" basis,
+# without warranties or conditions of any kind, either express or implied.
+# see the license for the specific language governing permissions and
+# limitations under the license.
+
+import unittest
+import os
+import sys
+import argparse
+import shutil
+import logging
+import struct
+import six
+import numpy as np
+import paddle
+import paddle.fluid as fluid
+from paddle.fluid.contrib.slim.core import Compressor
+
+logging.basicConfig(format='%(asctime)s-%(levelname)s: %(message)s')
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--batch_size', type=int, default=1, help='batch size')
+    parser.add_argument(
+        '--infer_model',
+        type=str,
+        default='',
+        help='path to fp32 programdesc with fp32 weights')
+    parser.add_argument('--infer_data', type=str, default='', help='data file')
+    parser.add_argument(
+        '--int8_model_save_path',
+        type=str,
+        default='./output',
+        help='path to save int8 programdesc with fp32 weights')
+    parser.add_argument(
+        '--warmup_batch_size',
+        type=int,
+        default=100,
+        help='batch size for quantization warmup')
+    parser.add_argument(
+        '--accuracy_diff_threshold',
+        type=float,
+        default=0.01,
+        help='accepted accuracy drop threshold.')
+
+    test_args, args = parser.parse_known_args(namespace=unittest)
+
+    return test_args, sys.argv[:1] + args
+
+
+class TestMKLDNNPostTrainingQuantStrategy(unittest.TestCase):
+    """
+    Test API of Post Training quantization strategy for int8 with MKL-DNN.
+    """
+
+    def _reader_creator(self,
+                        data_file='data.bin',
+                        cycle=False,
+                        predict_num=100):
```

Comment:
Please, add an option to set the number of images to process. Currently, the value of 10 is hardcoded in lines 199 and 203, and cannot be modified.

Reply:
OK. This can be done later.